### PR TITLE
use Pnpm replace Yarn

### PR DESCRIPTION
**Pull request by shixin-guo**
- Utilize the workspace protocol to import private packages, eliminating the need to change the version every time.
- Add "website" and "ee/tabby-ui" to the workspaces, eliminating the manual installation of dependencies in the child folders.

### CODE REVIEW

**Comment by shixin-guo** [comment]
I'm not sure if there are any specific requirements for choosing Yarn. Migrating directly might be a bit hasty, but currently, Pnpm does seem more user-friendly. I hope this PR doesn't cause any code hassles for the team. 😄😄😄

**Comment by wsxiaoys** [comment]
The real issue is our desire to embed the built JavaScript into each repository. While this process could undoubtedly be enhanced, it doesn't seem directly related to the repository manager

cc @icycodes to confirm

**Comment by shixin-guo** [comment]
> The real issue is our desire to embed the built JavaScript into each repository. While this process could undoubtedly be enhanced, it doesn't seem directly related to the repository manager
> 
> cc @icycodes to confirm

thanks ! you can set up the build script for your workspace in each JS package. now updating dependencies may require you to upgrade them in every project and run install and build processes for each. This is the main reasion I create the PR ,and hope it will optimze developer exprience..

**Comment by icycodes** [comment]
Hi, @shixin-guo 
Thank you for your PR!
I am testing this branch on my local env.

File: package.json [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-    "clients/tabby-agent",
-    "clients/vscode",
-    "clients/vim",
-    "clients/intellij"
-  ],
   "engines": {
     "node": ">=18"
-  }
+  },
+  "packageManager": "^pnpm@8.3.1"
```
Comment (icycodes) [inline package.json:6]: It should be `"packageManager": "pnpm@8.3.1"` ?
The prefix `^` seems to be causing `Usage Error: Unsupported package manager specification (^pnpm@8.3.1,^pnpm,8.3.1)`

Comment (shixin-guo) [reply package.json:6]: Certainly, actually I think this PR is still in progress, and  if we plan to migrate to pnpm we also need to  update the CI jobs scripts and documentation together, I'll continue working on it if you agree this enhancement

**Pull request closed without merging**